pkg/types/validation/installconfig.go

```diff
@@ -542,8 +543,13 @@ func validateNamedRepository(r string) error {
 		// in an error. Instead we will check whether the input is
 		// a valid hostname as a workaround.
 		if err == dockerref.ErrNameNotCanonical {
-			_, err := url.ParseRequestURI(r)
+			// If the hostname string contains a port, lets attempt
+			// to split them
+			host, _, err := net.SplitHostPort(r)
 			if err != nil {
+				host = r
+			}
+			if !govalidator.IsHost(host) {
```

Comment:
Can we use https://github.com/openshift/installer/blob/c88793ce714db6856284faa91542a2cfb7b51437/pkg/validate/validate.go#L275 (once https://github.com/openshift/installer/pull/5367 merges)? I would rather that than bring in another dependency.

Reply:
+1. Will change. Thanks.